src/client.ts

```diff
@@ -47,43 +47,45 @@ export class UXDClient {
     });
   }
 
-  public createSetRedeemableGlobalSupplyCapInstruction(
+  public createEditControllerInstruction(
     controller: Controller,
     authority: PublicKey,
-    supplyCapAmount: number,
+    uiFields: {
+      quoteMintAndRedeemSoftCap?: {
+        value: number;
+        depository: MangoDepository;
+      };
+      redeemableSoftCap?: number;
+      redeemableGlobalSupplyCap?: number;
+    },
     options: ConfirmOptions
-  ): TransactionInstruction {
-    const supplyCapAmountNativeUnits = uiToNative(
-      supplyCapAmount,
-      controller.redeemableMintDecimals
-    );
-    return this.instruction.setRedeemableGlobalSupplyCap(
-      supplyCapAmountNativeUnits,
+  ) {
+    const quoteMintAndRedeemSoftCap = uiFields.quoteMintAndRedeemSoftCap;
+    const redeemableSoftCap = uiFields.redeemableSoftCap;
+    const redeemableGlobalSupplyCap = uiFields.redeemableGlobalSupplyCap;
+    return this.instruction.editController(
       {
-        accounts: {
-          authority,
-          controller: controller.pda,
-        },
-        options: options,
-      }
-    );
-  }
-
-  public createSetMangoDepositoriesRedeemableSoftCapInstruction(
-    controller: Controller,
-    authority: PublicKey,
-    supplySoftCapAmount: number,
-    options: ConfirmOptions
-  ): TransactionInstruction {
-    const supplySoftCapAmountNativeUnits = uiToNative(
-      supplySoftCapAmount,
-      controller.redeemableMintDecimals
-    );
-    return this.instruction.setMangoDepositoriesRedeemableSoftCap(
-      supplySoftCapAmountNativeUnits,
+        quoteMintAndRedeemSoftCap: quoteMintAndRedeemSoftCap
+          ? uiToNative(
+              quoteMintAndRedeemSoftCap.value,
+              quoteMintAndRedeemSoftCap.depository.quoteMintDecimals // special case
+            )
+          : null,
+        redeemableSoftCap:
+          redeemableSoftCap !== undefined
```

Comment:
need to check for it being truthy (as above), or being `undefined` (here) ?

Reply:
I think here we need to check specifically if the value is not `undefined`, since a value of `0` is not truthy, but is a valid value